src/udev/udev-builtin-input_id.c

```diff
@@ -240,13 +241,18 @@ static bool test_pointers(struct udev_device *dev,
                         is_touchscreen = true;
         }
 
-        if (has_rel_coordinates && has_mouse_button)
-                is_mouse = true;
+        if (has_mouse_button)
+                if (has_rel_coordinates)
+                        is_mouse = true;
+                if (!has_abs_coordinates) /* exclude vmmouse and joysticks */
+                        is_pad = true;
 
         if (is_pointing_stick)
                 udev_builtin_add_property(dev, test, "ID_INPUT_POINTINGSTICK", "1");
         if (is_mouse)
                 udev_builtin_add_property(dev, test, "ID_INPUT_MOUSE", "1");
+        if (is_pad)
+                udev_builtin_add_property(dev, test, "ID_INPUT_MOUSE", "1");
```

Comment:
hmm, the two if checks above have the same body, hence they should be combined as `if (is_mouse || is_pad) …`